File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnHealBeforeMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnHealBeforeMessage implements IMessage{
    public static readonly Id = "OnHealBeforeMessage";
    public constructor(public source:AbstractCreature,public Amount : number){

    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnHealBeforeMessage.cs
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///受到治疗后
    public class OnHealBeforeMessage : IEventHandle
    {
        public AbstractCreature Unit;
        public int Amount;

        public OnHealBeforeMessage(AbstractCreature unit, int amount)
        {
            Unit = unit;
            Amount = amount;
        }
    }
}